app/components/base/breadcrumbs/breadcrumb.service.ts

```diff
@@ -0,0 +1,145 @@
+import { Injectable } from "@angular/core";
+import { ActivatedRoute, NavigationEnd, PRIMARY_OUTLET, Router } from "@angular/router";
+import { BehaviorSubject, Observable } from "rxjs";
+
+import { Constants } from "app/utils";
+import { Breadcrumb, RouteComponent } from "./breadcrumb-models";
+
+function breadcrumbMethodMessage(componentName) {
```

Comment:
Minor: may want to rename this to "breadcrumbMissingMethodMessage" or something similar.